### PR TITLE
Support shared flag

### PR DESCRIPTION
**Pull request by kmaragon**
Since building static or shared libs has become something of a standard in conan, it would be nice if this package had that as well.

### CODE REVIEW

**Comment by danimtb** [comment]
Hi @kmaragon,

Thanks for giving this a try.

I also tried making shared packages myself before but didn't manage to get it working. I think it is not possible specially for the mingw --> Visual Studio conversion.

Anyway, I will be happy to include your changes once CI passes. Some suggestions:

- Please stick to PEP8 for line lenght.
- Make sure gfortran system requirements packages are available for CI to install (see some grfortran installations failing in CI)
- Avoid setting -fPIC flag manually as much as possible. CMake helper already defines `CMAKE_POSITION_INDEPENDENT_CODE` if `fPIC=True` in options. Also make sure to delete this option for windows package as it does not make sense:

  ```
  def config_options(self):
      if self.settings.os == "Windows":
          del self.options.fPIC
  ```

**Comment by kmaragon** [comment]
Ok. But as for CI passing, how should I handle the fact that the builds timeout against travis' hard limits? Is there something I can tweak somewhere to have them give the build a few more mintes?

**Comment by danimtb** [comment]
Yes, I increased the number on pages in Travis to avoid hitting the time limit. Now it should be fine 😄

File: conanfile.py [hunk 1]
```diff
@@ -101,5 +120,18 @@ def package(self):
...
     def package_info(self):
-        self.cpp_info.libs = tools.collect_libs(self)
+        # the order is important for static builds
+        self.cpp_info.libs = ["lapacke", "lapack", "blas", "cblas", "gfortran"]
+        self.cpp_info.libdirs = ["lib"]
+        if tools.os_info.is_macos:
+            brewout = StringIO()
+            try:
+                self.run("brew ls --verbose gcc | grep libgfortran | grep {}".format(self.settings.arch), output=brewout)
+            except e as Exception:
```
Comment (danimtb) [inline conanfile.py]: This should be: ``except Exception as e:``

**Comment by danimtb** [comment]
Thanks for the help pushing this forward @SSE4 !

**Pull request merged**